org.code.toboggan.network/src/org/code/toboggan/network/NetworkActivator.java

```diff
@@ -17,6 +18,8 @@ static BundleContext getContext() {
 	 */
 	public void start(BundleContext bundleContext) throws Exception {
 		NetworkActivator.context = bundleContext;
+		// Start WS connection when this plugin is activated
+		WSService.getWSManager().connect();
```

Comment:
The order of this call would be getWSManager() -> connect(), no? And getWSManager calls the constructor for the service and registers the listeners, so the order should be fine.